spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/ModuleDefinition.java

```diff
@@ -0,0 +1,397 @@
+	 */
+	private final String group;
+
+	/**
+	 * Position in stream definition relative to the other modules
+	 * in the definition. 0 indicates the first/leftmost position.
+	 */
+	private final int index;
+
+	/**
+	 * Map of channel bindings for this module (such as input and/or output channels).
+	 */
+	private final Map<String, String> bindings;
+
+	/**
+	 * Parameters for module. This is specific to the type of module - for
+	 * instance an http module would include a port number as a parameter.
+	 */
+	private final Map<String, String> parameters;
+
+
+	/**
+	 * Construct a {@code ModuleDefinition}. This constructor is private; use
+	 * {@link ModuleDefinition.Builder} to create a new instance.
+	 *
+	 * @param name name of module
+	 * @param label label used for module in stream definition
+	 * @param group group this module belongs to (stream)
+	 * @param index position of module in stream definition
+	 * @param bindings map of channel bindings; may not be {@code null}
+	 * @param parameters module parameters; may be {@code null}
+	 */
+	private ModuleDefinition(String name, String label, String group,
+			int index, Map<String, String> bindings, Map<String, String> parameters) {
+		Assert.notNull(name, "name must not be null");
+		Assert.notNull(label, "label must not be null");
+		Assert.notNull(group, "group must not be null");
+		Assert.notNull(bindings, "bindings must not be null");
+		Assert.notEmpty(bindings, "bindings must not be empty");
+		this.name = name;
+		this.label = label;
+		this.group = group;
+		this.index = index;
+		this.bindings = Collections.unmodifiableMap(new HashMap<String, String>(bindings));
+		this.parameters = parameters == null
+				? Collections.<String, String>emptyMap()
+				: Collections.unmodifiableMap(new HashMap<String, String>(parameters));
+	}
+
+	/**
+	 * Return the name of this module.
+	 *
+	 * @return module name
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Return symbolic name of a module. This may be explicitly specified in
+	 * the DSL string, which is required if the stream contains another module
+	 * with the same name. Otherwise, it will be the same as the module name.
+	 *
+	 * @return module label
+	 */
+	public String getLabel() {
+		return label;
+	}
+
+	/**
+	 * Return name of deployable unit this module instance belongs to
+	 * (such as a stream).
+	 *
+	 * @return group name
+	 */
+	public String getGroup() {
+		return group;
+	}
+
+	/**
+	 * Return position in stream definition relative to the other
+	 * modules in the definition. 0 indicates the first/leftmost position.
+	 *
+	 * @return module index
+	 */
+	public int getIndex() {
+		return index;
+	}
+
+	/**
+	 * Return map of channel bindings for this module (such as input
+	 * and/or output channels).
+	 *
+	 * @return map of channel bindings for module
+	 */
+	public Map<String, String> getBindings() {
+		return this.bindings;
```

Comment:
I generally feel like in the case of a getter it's unnecessary noise (a setter is different though, since the param name usually matches).